src/components/Tool/FixEmailAddresses/FixEmailAddresses.test.tsx

```diff
@@ -115,28 +124,147 @@ describe('FixPhoneNumbers-Home', () => {
     expect(getByTestId('starIcon-testid2-0')).toBeInTheDocument();
   });
 
-  //TODO: Fix during MPDX-7946
-  it.skip('add an email address to second person', async () => {
-    const { getByTestId, getByDisplayValue } = render(<Components />);
-    await waitFor(() =>
-      expect(getByTestId('starIcon-testid2-0')).toBeInTheDocument(),
-    );
-    expect(getByTestId('textfield-testid2-0')).toBeInTheDocument();
-
-    const textfieldNew1 = getByTestId(
-      'addNewEmailInput-testid2',
-    ) as HTMLInputElement;
-    userEvent.type(textfieldNew1, 'email12345@gmail.com');
-    const addButton1 = getByTestId('addButton-testid2');
-    userEvent.click(addButton1);
-
-    expect(textfieldNew1.value).toBe('');
-    expect(getByTestId('textfield-testid2-2')).toBeInTheDocument();
-    expect(getByDisplayValue('email12345@gmail.com')).toBeInTheDocument();
+  describe('add email address', () => {
+    interface AddEmailAddressProps {
+      postSaveResponse: object;
+      emailAddressNodes: object[];
+      elementToWaitFor: string;
+      textFieldIndex: number;
+      addButtonId: string;
+      cache: InMemoryCache;
+    }
+
+    const addEmailAddress = async ({
+      postSaveResponse,
+      emailAddressNodes,
+      elementToWaitFor,
+      textFieldIndex,
+      addButtonId,
+      cache,
+    }: AddEmailAddressProps) => {
+      let cardinality = 0;
+      jest.spyOn(cache, 'readQuery').mockReturnValue(postSaveResponse);
+      jest.spyOn(cache, 'writeQuery');
+
+      const updatePerson = {
+        person: {
+          emailAddresses: {
+            nodes: [
+              ...emailAddressNodes,
+              {
+                email: newEmail.email,
+              },
+            ],
+          },
+        },
+      } as ErgonoMockShape;
+
+      const { getByTestId, getAllByLabelText } = render(
+        <Components
+          mocks={{
+            GetInvalidEmailAddresses: () => {
+              let queryResult;
+              if (cardinality === 0) {
+                queryResult = {
+                  people: {
+                    nodes: mockInvalidEmailAddressesResponse,
+                  },
+                };
+              } else {
+                queryResult = postSaveResponse;
+              }
+              cardinality++;
+              return queryResult;
+            },
+            EmailAddresses: { updatePerson },
+          }}
+          cache={cache}
+        />,
+      );
+      await waitFor(() => {
+        expect(getByTestId(elementToWaitFor)).toBeInTheDocument();
+      });
+
+      const textFieldNew =
+        getAllByLabelText('New Email Address')[textFieldIndex];
+      userEvent.type(textFieldNew, newEmail.email);
+      const addButton = getByTestId(addButtonId);
+      userEvent.click(addButton);
+    };
+
+    it('should add an email address to the first person', async () => {
```

Comment:
We should make the test like how the user would interact with the component. Such as types in new email and clicks "add", this should fire a GraphQL Mutation request we can test.

Reply:
That is what we are doing here, though I never found good ways to test what data got sent in the mutation, so I went with what changed on the screen as most other tests I've found are doing.

Reply:
I have updated the code to add some more tests and clean up the PR.